### PR TITLE
syscall filter group updates

### PR DESCRIPTION
**Pull request by poettering**
Playing a bit of catchup and a fix for #26913

### CODE REVIEW

File: src/shared/seccomp-util.c [hunk 1]
```diff
@@ -797,9 +797,19 @@ const SyscallFilterSet syscall_filter_sets[_SYSCALL_FILTER_SET_MAX] = {
                 "sched_setparam\0"
                 "sched_setscheduler\0"
                 "set_mempolicy\0"
+                "set_mempolicy_home_node\0"
```
Comment (bluca) [inline src/shared/seccomp-util.c:802]: this seems unrelated to the commit?

File: src/analyze/analyze-syscall-filter.c [hunk 2]
```diff
@@ -58,6 +58,26 @@ static int load_kernel_syscalls(Set **ret) {
...
 
+static int syscall_set_add(Set **s, const SyscallFilterSet *set) {
+        int r;
+
+        assert(s);
+
+        if (!set)
+                return 0;
+
+        NULSTR_FOREACH(syscall, set->value) {
```
Comment (bluca) [inline src/analyze/analyze-syscall-filter.c]: `syscall` is a function name, maybe better to pick a different variable name for the iterator to avoid ambiguity? Github review highlights it in a scary red color!

**Comment by poettering** [comment]
updated, ptal

**Review by bluca — APPROVED** [review]

**Review by yuwata — APPROVED** [review]
LGTM.

File: src/analyze/analyze-syscall-filter.c [hunk 3]
```diff
@@ -58,15 +58,35 @@ static int load_kernel_syscalls(Set **ret) {
...
+        int r;
+
+        assert(s);
+
+        if (!set)
+                return 0;
+
+        NULSTR_FOREACH(sc, set->value) {
+                if (sc[0] == '@')
+                        continue;
```
Comment (keszybz) [inline src/analyze/analyze-syscall-filter.c:71]: Shouldn't this add `sc` recursively?

Comment (poettering) [reply src/analyze/analyze-syscall-filter.c:71]: no. this is called from a loop around all groups. so if we follow the groups here we'd add them multiple times, which is pointless.
